### PR TITLE
enable npm provenance

### LINKED ISSUE #63: Enable NPM Provenance
https://github.blog/2023-04-19-introducing-npm-package-provenance/

The old packages had it enabled:
image

it's here:
https://www.npmjs.com/package/@shopware-ag/meteor-admin-sdk

it's missing:
https://www.npmjs.com/package/@shopware-ag/meteor-icon-kit

### PR DESCRIPTION
**Pull request by Haberkamp**
## What?

This PR enables [provenance](https://github.blog/2023-04-19-introducing-npm-package-provenance/) for our npm packages.

## Why?

This increases trust for our packages, because we can be sure that the packages indeed published by us and not by a malicious person.

## How?

I've added a `NPM_CONFIG_PROVENANCE` environment variable to the CI pipeline. 

## Testing?

The changes will only have effect once this PR gets merged and we do another release.

## Anything Else?

Closes #63

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -43,3 +43,4 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NPM_CONFIG_PROVENANCE: ${{ secrets.NPM_CONFIG_PROVENANCE }}
```
Comment (shyim) [inline .github/workflows/release.yml:46]: why an secret for this?

Comment (shyim) [reply .github/workflows/release.yml:46]:
```
permissions:
      contents: read
      id-token: write
```

is missing for oidc connection

Comment (Haberkamp) [reply .github/workflows/release.yml:46]: Oh yeah, we could just set it there directly.

> permissions:
      contents: read
      id-token: write
      
> is missing for oidc connection

Could you commit the appropiate changes? I think that will be faster as you're have more knowledge in that area.

Comment (Haberkamp) [reply .github/workflows/release.yml]: Reading the [npm documentation](https://docs.npmjs.com/generating-provenance-statements#using-third-party-package-publishing-tools) this should be enough.

**Review by seggewiss — APPROVED** [review]

**Pull request merged**